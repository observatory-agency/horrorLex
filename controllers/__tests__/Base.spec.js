const BookModel = require('../../models/Book');
const BaseController = require('../Base');

jest.mock('../../models/Book');

class SomeOtherController extends BaseController {
  constructor() {
    super();
    this.get = 'get';
    this.post = 'post';
  }

  get() {
    return this.get;
  }

  post() {
    return this.post;
  }
}

describe('BaseController', () => {
  let base;
  beforeEach(() => {
    base = new BaseController();
    someOtherController = new SomeOtherController();
  });

  describe('constructor', () => {
    it('should set up a this.modelMap', () => {
      expect(base.modelMap).toBeDefined();
    });
    it('should set this.get', () => {
      expect(base.get).toBeDefined();
    });
    it('should set this.post', () => {
      expect(base.post).toBeDefined();
    });
  });

  describe('model', () => {
    it('should return the modelMap with instantiated models', () => {
      expect(base.model).toMatchSnapshot();
    });
  });
});
