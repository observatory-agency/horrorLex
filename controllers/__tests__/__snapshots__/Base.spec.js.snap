// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseController model should return the modelMap with instantiated models 1`] = `
Object {
  "book": BookModel {
    "aggregate": [MockFunction],
    "find": [MockFunction],
    "findOne": [MockFunction],
  },
}
`;
